NickvisionTubeConverter.GNOME/Controls/HistoryDialog.cs

```diff
@@ -59,17 +70,31 @@ private HistoryDialog(Gtk.Builder builder, Gtk.Window parent, string iconName, D
             }
             row.SetTitleLines(1);
             row.SetSubtitleLines(1);
-            var button = Gtk.Button.New();
-            button.SetIconName("view-refresh-symbolic");
-            button.SetTooltipText(_("Download Again"));
-            button.SetValign(Gtk.Align.Center);
-            button.AddCssClass("flat");
-            button.OnClicked += (sender, e) =>
+            var downloadButton = Gtk.Button.New();
+            downloadButton.SetIconName("view-refresh-symbolic");
+            downloadButton.SetTooltipText(_("Download Again"));
+            downloadButton.SetValign(Gtk.Align.Center);
+            downloadButton.AddCssClass("flat");
+            downloadButton.OnClicked += (sender, e) =>
             {
                 DownloadAgainRequested?.Invoke(this, pair.Key);
             };
-            row.AddSuffix(button);
-            row.SetActivatableWidget(button);
+            row.AddSuffix(downloadButton);
+            if(File.Exists(pair.Value.Path))
+            {
+                var openButton = Gtk.Button.New();
+                openButton.SetIconName("document-open-symbolic");
+                openButton.SetTooltipText(_("Open"));
```

Comment:
Does it open file or folder with file selected? If the former, then "play" icon would be better. If the latter, I would change tooltip to "Open in Folder".

Reply:
it's the former so I will change to play icon